packages/happn-3/test/__fixtures/test/integration/security/authentication/workingAuth.js

```diff
@@ -7,25 +9,32 @@ module.exports = class TestAuthProvider extends require('../../../../../..').pro
     return new TestAuthProvider(securityFacade, config, providerOptions);
   }
 
-  async providerCredsLogin(credentials, sessionId) {      
-    if (credentials.username === "secondTestuser@somewhere.com" && credentials.password === "secondPass") {        
-      let user = {username: "secondTestuser@somewhere.com", groups:[]}
+  async providerCredsLogin(credentials, sessionId) {
+    if (
+      credentials.username === 'secondTestuser@somewhere.com' &&
+      credentials.password === 'secondPass'
+    ) {
+      let user = { username: 'secondTestuser@somewhere.com', groups: [] };
       return this.loginOK(credentials, user, sessionId);
     }
     return this.loginFailed(credentials.username, 'Invalid credentials');
   }
 
-  async providerTokenLogin(credentials, decodedToken, sessionId) {      
-    if (decodedToken.username === "secondTestuser@somewhere.com" && credentials.token != null) {        
-      let user = {username: "secondTestuser@somewhere.com", groups:[]}
+  async providerTokenLogin(credentials, decodedToken, sessionId) {
+    if (decodedToken.username === 'secondTestuser@somewhere.com' && credentials.token != null) {
+      let user = { username: 'secondTestuser@somewhere.com', groups: [] };
       return this.loginOK(credentials, user, sessionId);
     }
     return this.loginFailed(credentials.username, 'Invalid credentials');
   }
 
+  async providerResetPassword(credentials) {
+    throw new Error(`Works !! Password reset ${credentials.username}`);
```

Comment:
I assume this is so you can test both the bubbling of an error and ensure the plugin method gets reached?

Reply:
Checking we get the username too. This in a auth provider will probably just run with no errors on success.